### PR TITLE
e2e: Embed DNS related maninfests into codebase

### LINKED ISSUE #94227: example_cluster_dns.go relies on GOPATH, examples/ files
**What happened**:

We've been doing some release readiness and auditing all `sig-network` tagged ginkgo tests...  Looks like The example_cluster_dns.go test seems to expect a GOPATH when running.  Im seeing some failures in sonobuoy and suspect a possible relation there but havent dove into it.

Either way, its clear this test might need some updating/fixing/cleanuping and parts can be rewritten/improved.

**What you expected to happen**:

- No reliance on GOPATH for an e2e test

... I dont remember the whole history on this, but no tests that rely on static files currently look in gopath,  from what i can tell ?  instead we plumbed in a repo-root/ option a few years ago.  There may be a state of the art in terms of how we do this.... but i thought it was `repo-root` at one point which was an option passed in.

**How to reproduce it (as minimally and precisely as possible)**:

Try to run this in sonobuoy or any other environment where there's no GOPATH or non-standard source trees, pretty sure it might not work. 

**Anything else we need to know?**:

I think the fix here is to just embed the YAML in golang like we do in other tests. along the way there's other cleanup we can do to improve this test.

**Environment**:
- Kubernetes version (use `kubectl version`): master

### PR DESCRIPTION
**Pull request by chendave**
#### What type of PR is this?
/kind bug

#### Which issue(s) this PR fixes:

Fixes #94227

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by aojea** [comment]
I can't find any job running this test, are these test running somewhere? 
Can you verify the test work after the changes?

**Comment by chendave** [comment]
@aojea 

yes, it works, I have verified this locally.

```
# /usr/bin/e2e.test -ginkgo.focus="\[sig-network\] ClusterDns \[Feature:Example\] should create pod that uses dns"  -provider=local

{"msg":"PASSED [sig-network] ClusterDns [Feature:Example] should create pod that uses dns","total":1,"completed":1,"skipped":5967,"failed":0}
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSFeb 24 10:56:55.459: INFO: Running AfterSuite actions on all nodes
Feb 24 10:56:55.459: INFO: Running Cleanup Action: k8s.io/kubernetes/test/e2e/storage.glob..func18.2
Feb 24 10:56:55.459: INFO: Running Cleanup Action: k8s.io/kubernetes/test/e2e/storage.glob..func8.2
Feb 24 10:56:55.459: INFO: Running Cleanup Action: k8s.io/kubernetes/test/e2e/storage.glob..func7.2
Feb 24 10:56:55.459: INFO: Running Cleanup Action: k8s.io/kubernetes/test/e2e/storage/vsphere.glob..func17.3
Feb 24 10:56:55.459: INFO: Running Cleanup Action: k8s.io/kubernetes/test/e2e/storage/vsphere.glob..func9.2
Feb 24 10:56:55.459: INFO: Running Cleanup Action: k8s.io/kubernetes/test/e2e/storage/vsphere.glob..func4.2
Feb 24 10:56:55.459: INFO: Running Cleanup Action: k8s.io/kubernetes/test/e2e/storage/vsphere.glob..func1.3
Feb 24 10:56:55.459: INFO: Running AfterSuite actions on node 1
{"msg":"Test Suite completed","total":1,"completed":1,"skipped":7115,"failed":0}

Ran 1 of 7116 Specs in 13.244 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 7115 Skipped

```

File: test/e2e/testing-manifests/cluster-dns/dns-frontend-pod.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  labels:
+    name: dns-frontend
+spec:
+  containers:
+    - name: dns-frontend
+      image: k8s.gcr.io/example-dns-frontend:v1
+      command:
+        - python
+        - client.py
+        - http://dns-backend.development.svc.cluster.local:8000
```
Comment (aojea) [inline test/e2e/testing-manifests/cluster-dns/dns-frontend-pod.yaml:14]: We can't assume cluster.local is the domain in all clusters, I think this should use .svc as top domain

Comment (aojea) [reply test/e2e/testing-manifests/cluster-dns/dns-frontend-pod.yaml:14]: Ah, it is replaced during the test
Scratch this comment

**Comment by aojea** [comment]
/lgtm
/approve 

thanks

File: test/e2e/testing-manifests/cluster-dns/dns-frontend-pod.yaml (hunk 1 above)
Comment (chendave) [reply test/e2e/testing-manifests/cluster-dns/dns-frontend-pod.yaml:14]: thanks for the detailed review, anyway! :)

**Pull request merged**